app/dashboard.rb

```diff
@@ -114,11 +74,86 @@ def options(input)
     end
   end
 
+  def save_data
+    FileUtils.mkdir_p('./app_data/')
+    FileUtils.cd('./app_data/') do
+      FileUtils.rm_f('books.json')
+      FileUtils.rm_f('people.json')
+      FileUtils.rm_f('rentals.json')
+      FileUtils.touch('books.json')
+      FileUtils.touch('people.json')
+      FileUtils.touch('rentals.json')
+
+      # generate json object
+      books_json = []
+      @my_app.books.each { |book| books_json << JSON.generate(book.export_json) }
+      people_json = []
+      @my_app.people.each { |person| people_json << JSON.generate(person.export_json) }
+      rentals_json = []
+      @my_app.rentals.each { |rental| rentals_json << JSON.generate(rental.export_json) }
+
+      # write data to their respective files
+      File.write('books.json', books_json)
+      File.write('people.json', people_json)
+      File.write('rentals.json', rentals_json)
+    end
+  end
```

Comment:
- [ ] You should handle scenarios for when the files do not exist. When I remove all storage files and start your app. After all data creation when I quit the app the files should be created and the data should be stored in them but that is not the case now. Kindly implement that.

Reply:
@julie-ify Kindly explain a little more about this which file do you mean??

Reply:
well resolved! ✔️